diff -Naurp r8168-8.048.03.orig/src/r8168_n.c r8168-8.048.03/src/r8168_n.c
--- r8168-8.048.03.orig/src/r8168_n.c	2020-04-13 09:41:16.000000000 +0100
+++ r8168-8.048.03/src/r8168_n.c	2021-03-21 17:22:54.504574470 +0000
@@ -62,9 +62,11 @@
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
 #if LINUX_VERSION_CODE < KERNEL_VERSION(5,4,0)
+#if RHEL_MAJOR != 8
 #include <linux/pci-aspm.h>
 #endif
 #endif
+#endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,37)
 #include <linux/prefetch.h>
 #endif
@@ -456,7 +458,7 @@ static void rtl8168_hw_config(struct net
 static void rtl8168_hw_start(struct net_device *dev);
 static int rtl8168_close(struct net_device *dev);
 static void rtl8168_set_rx_mode(struct net_device *dev);
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0) || RHEL_MAJOR >= 8
 static void rtl8168_tx_timeout(struct net_device *dev, unsigned int txqueue);
 #else
 static void rtl8168_tx_timeout(struct net_device *dev);
@@ -27943,7 +27945,7 @@ static void rtl8168_reset_task(struct wo
         }
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0) || RHEL_MAJOR >= 8
 static void
 rtl8168_tx_timeout(struct net_device *dev, unsigned int txqueue)
 #else
