diff -aurp r8168-8.055.00.orig/src/r8168.h r8168-8.055.00/src/r8168.h
--- r8168-8.055.00.orig/src/r8168.h	2025-01-15 14:24:13.000000000 +0000
+++ r8168-8.055.00/src/r8168.h	2025-03-01 16:10:48.587577006 +0000
@@ -165,7 +165,7 @@ static inline void netdev_tx_completed_q
 static inline void netdev_tx_reset_queue(struct netdev_queue *q) {}
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(5,8,0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,8,0) && RHEL_MAJOR < 8
 static inline void fsleep(unsigned long usecs)
 {
         if (usecs <= 10)
@@ -751,7 +751,7 @@ typedef int *napi_budget;
 typedef struct napi_struct *napi_ptr;
 typedef int napi_budget;
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,1,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,1,0) || RHEL_MAJOR >=8
 #define RTL_NAPI_CONFIG(ndev, priv, function, weight)   netif_napi_add_weight(ndev, &priv->napi, function, weight)
 #else
 #define RTL_NAPI_CONFIG(ndev, priv, function, weight)   netif_napi_add(ndev, &priv->napi, function, weight)
diff -aurp r8168-8.055.00.orig/src/r8168_n.c r8168-8.055.00/src/r8168_n.c
--- r8168-8.055.00.orig/src/r8168_n.c	2025-01-15 14:24:13.000000000 +0000
+++ r8168-8.055.00/src/r8168_n.c	2025-03-01 16:08:49.457529898 +0000
@@ -63,9 +63,11 @@
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
 #if LINUX_VERSION_CODE < KERNEL_VERSION(5,4,0)
+#if RHEL_MAJOR != 8
 #include <linux/pci-aspm.h>
 #endif
 #endif
+#endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,37)
 #include <linux/prefetch.h>
 #endif
@@ -591,7 +593,7 @@ static irqreturn_t rtl8168_interrupt(int
 #endif
 static void rtl8168_rx_desc_offset0_init(struct rtl8168_private *, int);
 static void rtl8168_set_rx_mode(struct net_device *dev);
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0) || RHEL_MAJOR >=8
 static void rtl8168_tx_timeout(struct net_device *dev, unsigned int txqueue);
 #else
 static void rtl8168_tx_timeout(struct net_device *dev);
@@ -7256,7 +7258,7 @@ rtl8168_set_ring_size(struct rtl8168_pri
 }
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,17,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,17,0) || RHEL_MAJOR >= 8
 static void rtl8168_get_ringparam(struct net_device *dev,
                                   struct ethtool_ringparam *ring,
                                   struct kernel_ethtool_ringparam *kernel_ring,
@@ -7274,7 +7276,7 @@ static void rtl8168_get_ringparam(struct
         ring->tx_pending = tp->tx_ring[0].num_tx_desc;
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,17,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,17,0) || RHEL_MAJOR >= 8
 static int rtl8168_set_ringparam(struct net_device *dev,
                                  struct ethtool_ringparam *ring,
                                  struct kernel_ethtool_ringparam *kernel_ring,
@@ -28716,7 +28718,7 @@ rtl8168_init_one(struct pci_dev *pdev,
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,22)
                 if ((tp->mcfg == CFG_METHOD_1) || (tp->mcfg == CFG_METHOD_2) || (tp->mcfg == CFG_METHOD_3)) {
                         dev->hw_features &= ~NETIF_F_IPV6_CSUM;
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,19,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,19,0) || RHEL_MAJOR >= 8
                         netif_set_tso_max_size(dev, LSO_64K);
                         netif_set_tso_max_segs(dev, NIC_MAX_PHYS_BUF_COUNT_LSO2);
 #else //LINUX_VERSION_CODE >= KERNEL_VERSION(5,19,0)
@@ -28742,7 +28744,7 @@ rtl8168_init_one(struct pci_dev *pdev,
                                         break;
                                 };
                         }
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,19,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,19,0) || RHEL_MAJOR >= 8
                         netif_set_tso_max_size(dev, LSO_64K);
                         netif_set_tso_max_segs(dev, NIC_MAX_PHYS_BUF_COUNT_LSO2);
 #else //LINUX_VERSION_CODE >= KERNEL_VERSION(5,19,0)
@@ -31092,7 +31094,7 @@ out_unlock:
         rtnl_unlock();
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,6,0) || RHEL_MAJOR >= 8
 static void
 rtl8168_tx_timeout(struct net_device *dev, unsigned int txqueue)
 #else
