--- a/e100.c	2014-08-06 01:54:27.000000000 +0100
+++ b/e100.c	2018-04-14 20:53:45.509052073 +0100
@@ -2834,7 +2834,7 @@ static const struct net_device_ops e100_
 	.ndo_validate_addr	= eth_validate_addr,
 	.ndo_set_rx_mode	= e100_set_multicast_list,
 	.ndo_set_mac_address	= e100_set_mac_address,
-	.ndo_change_mtu		= e100_change_mtu,
+	.ndo_change_mtu_rh74	= e100_change_mtu,
 	.ndo_do_ioctl		= e100_do_ioctl,
 	.ndo_tx_timeout		= e100_tx_timeout,
 #ifdef CONFIG_NET_POLL_CONTROLLER
