--- mptspi.c.orig	2022-04-14 16:51:12.000000000 +0100
+++ mptspi.c	2022-05-18 17:47:12.898090593 +0100
@@ -1238,7 +1238,7 @@ static struct spi_function_template mpts
  */
 
 static struct pci_device_id mptspi_pci_table[] = {
-#ifdef CONFIG_RHEL_DIFFERENCES
+#ifdef CONFIG_ELREPO_DIFFERENCES
 	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_53C1030,
 	PCI_VENDOR_ID_VMWARE, PCI_ANY_ID },
 #else
@@ -1540,7 +1540,7 @@ mptspi_probe(struct pci_dev *pdev, const
 
 	scsi_scan_host(sh);
 
-#ifdef CONFIG_RHEL_DIFFERENCES
+#ifdef CONFIG_ELREPO_DIFFERENCES
 	add_taint(TAINT_SUPPORT_REMOVED, LOCKDEP_STILL_OK);
 	pr_warn("MPTSPI MODULE IS NOT SUPPORTED\n");
 #endif
