diff -Naurp via-velocity-1.15.orig/via-velocity.c via-velocity-1.15/via-velocity.c
--- via-velocity-1.15.orig/via-velocity.c	2021-11-21 12:49:09.000000000 +0000
+++ via-velocity-1.15/via-velocity.c	2023-05-09 17:05:24.968114986 +0100
@@ -2844,8 +2844,7 @@ static int velocity_probe(struct device
 
 	netdev->netdev_ops = &velocity_netdev_ops;
 	netdev->ethtool_ops = &velocity_ethtool_ops;
-	netif_napi_add(netdev, &vptr->napi, velocity_poll,
-							VELOCITY_NAPI_WEIGHT);
+	netif_napi_add(netdev, &vptr->napi, velocity_poll);
 
 	netdev->hw_features = NETIF_F_IP_CSUM | NETIF_F_SG |
 			   NETIF_F_HW_VLAN_CTAG_TX;
