diff -Naurp iwlegacy-0.0.orig/3945-mac.c iwlegacy-0.0/3945-mac.c
--- iwlegacy-0.0.orig/3945-mac.c	2023-05-11 15:04:52.000000000 +0100
+++ iwlegacy-0.0/3945-mac.c	2023-05-16 20:32:57.509164400 +0100
@@ -1372,9 +1372,9 @@ il3945_dump_nic_error_log(struct il_priv
 }
 
 static void
-il3945_irq_tasklet(struct tasklet_struct *t)
+il3945_irq_tasklet(unsigned long data)
 {
-	struct il_priv *il = from_tasklet(il, t, irq_tasklet);
+	struct il_priv *il = (struct il_priv *)data;
 	u32 inta, handled = 0;
 	u32 inta_fh;
 	unsigned long flags;
@@ -3399,7 +3399,9 @@ il3945_setup_deferred_work(struct il_pri
 
 	timer_setup(&il->watchdog, il_bg_watchdog, 0);
 
-	tasklet_setup(&il->irq_tasklet, il3945_irq_tasklet);
+	tasklet_init(&il->irq_tasklet,
+		     il3945_irq_tasklet,
+		     (unsigned long)il);
 
 	return 0;
 }
diff -Naurp iwlegacy-0.0.orig/4965-mac.c iwlegacy-0.0/4965-mac.c
--- iwlegacy-0.0.orig/4965-mac.c	2023-05-11 15:04:52.000000000 +0100
+++ iwlegacy-0.0/4965-mac.c	2023-05-16 20:31:03.353430015 +0100
@@ -4338,9 +4338,9 @@ il4965_synchronize_irq(struct il_priv *i
 }
 
 static void
-il4965_irq_tasklet(struct tasklet_struct *t)
+il4965_irq_tasklet(unsigned long data)
 {
-	struct il_priv *il = from_tasklet(il, t, irq_tasklet);
+	struct il_priv *il = (struct il_priv *)data;
 	u32 inta, handled = 0;
 	u32 inta_fh;
 	unsigned long flags;
@@ -6234,7 +6234,9 @@ il4965_setup_deferred_work(struct il_pri
 
 	timer_setup(&il->watchdog, il_bg_watchdog, 0);
 
-	tasklet_setup(&il->irq_tasklet, il4965_irq_tasklet);
+	tasklet_init(&il->irq_tasklet,
+		     il4965_irq_tasklet,
+		     (unsigned long)il);
 
 	return 0;
 }
