diff -Naurp a/typhoon.c b/typhoon.c
--- a/typhoon.c	2016-07-24 19:37:48.000000000 +0100
+++ b/typhoon.c	2018-04-14 21:12:44.617220794 +0100
@@ -2262,7 +2262,7 @@ static const struct net_device_ops typho
 	.ndo_get_stats		= typhoon_get_stats,
 	.ndo_validate_addr	= eth_validate_addr,
 	.ndo_set_mac_address	= eth_mac_addr,
-	.ndo_change_mtu		= eth_change_mtu,
+	.ndo_change_mtu_rh74	= eth_change_mtu,
 };
 
 static int
