diff -Naurp r8125-9.016.01.orig/src/r8125_n.c r8125-9.016.01/src/r8125_n.c
--- r8125-9.016.01.orig/src/r8125_n.c	2025-07-28 10:08:00.000000000 -0400
+++ r8125-9.016.01/src/r8125_n.c	2025-09-16 14:59:18.948966128 -0400
@@ -16465,6 +16465,7 @@ err_out:
         goto out;
 }
 
+#ifndef ENABLE_USE_FIRMWARE_FILE
 static bool
 rtl8125_test_phy_ocp_v4(struct rtl8125_private *tp)
 {
@@ -16614,6 +16615,7 @@ exit:
 
         return restore;
 }
+#endif
 
 static bool
 rtl8125_test_phy_ocp(struct rtl8125_private *tp)
@@ -16626,6 +16628,7 @@ rtl8125_test_phy_ocp(struct rtl8125_priv
         if (tp->TestPhyOcpReg == FALSE)
                 goto unlock;
 
+#ifndef ENABLE_USE_FIRMWARE_FILE
         switch (tp->HwSuppEsdVer) {
         case 4:
                 reset = rtl8125_test_phy_ocp_v4(tp);
@@ -16636,6 +16639,7 @@ rtl8125_test_phy_ocp(struct rtl8125_priv
         default:
                 goto unlock;
         }
+#endif
 
 unlock:
         r8125_spin_unlock(&tp->phy_lock, flags);
