diff -Naurp r8125-9.018.00.orig/src/r8125_ptp.c r8125-9.018.00/src/r8125_ptp.c
--- r8125-9.018.00.orig/src/r8125_ptp.c	2026-06-02 05:58:28.000000000 -0400
+++ r8125-9.018.00/src/r8125_ptp.c	2026-07-02 23:59:59.047851694 -0400
@@ -824,7 +824,7 @@ void rtl8125_ptp_init(struct rtl8125_pri
         switch (tp->HwSuppPtpVer) {
         case 3:
                 tp->pps_enable = 0;
-#if LINUX_VERSION_CODE < KERNEL_VERSION(6,13,0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,13,0) && !(RHEL_MAJOR == 10 && RHEL_MINOR >= 2)
                 hrtimer_init(&tp->pps_timer, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
                 tp->pps_timer.function = rtl8125_phy_hrtimer_for_pps;
 #else
