diff -Naurp r8125-9.016.00.orig/src/r8125_n.c r8125-9.016.00/src/r8125_n.c
--- r8125-9.016.00.orig/src/r8125_n.c	2025-06-03 04:40:49.000000000 -0400
+++ r8125-9.016.00/src/r8125_n.c	2025-07-31 20:31:02.731060529 -0400
@@ -16458,6 +16458,7 @@ err_out:
         goto out;
 }
 
+#ifndef ENABLE_USE_FIRMWARE_FILE
 static bool
 rtl8125_test_phy_ocp_v4(struct rtl8125_private *tp)
 {
@@ -16607,6 +16608,7 @@ exit:
 
         return restore;
 }
+#endif
 
 static bool
 rtl8125_test_phy_ocp(struct rtl8125_private *tp)
@@ -16619,6 +16621,7 @@ rtl8125_test_phy_ocp(struct rtl8125_priv
         if (tp->TestPhyOcpReg == FALSE)
                 goto unlock;
 
+#ifndef ENABLE_USE_FIRMWARE_FILE
         switch (tp->HwSuppEsdVer) {
         case 4:
                 reset = rtl8125_test_phy_ocp_v4(tp);
@@ -16629,6 +16632,7 @@ rtl8125_test_phy_ocp(struct rtl8125_priv
         default:
                 goto unlock;
         }
+#endif
 
 unlock:
         r8125_spin_unlock(&tp->phy_lock, flags);
