diff -Naurp r8125-9.016.01.orig/src/r8125_n.c r8125-9.016.01.el10_2/src/r8125_n.c
--- r8125-9.016.01.orig/src/r8125_n.c	2025-07-28 10:08:00.000000000 -0400
+++ r8125-9.016.01.el10_2/src/r8125_n.c	2026-05-22 11:02:30.495285866 -0400
@@ -16465,6 +16465,7 @@ err_out:
         goto out;
 }
 
+#ifndef ENABLE_USE_FIRMWARE_FILE
 static bool
 rtl8125_test_phy_ocp_v4(struct rtl8125_private *tp)
 {
@@ -16614,6 +16615,7 @@ exit:
 
         return restore;
 }
+#endif
 
 static bool
 rtl8125_test_phy_ocp(struct rtl8125_private *tp)
@@ -16626,6 +16628,7 @@ rtl8125_test_phy_ocp(struct rtl8125_priv
         if (tp->TestPhyOcpReg == FALSE)
                 goto unlock;
 
+#ifndef ENABLE_USE_FIRMWARE_FILE
         switch (tp->HwSuppEsdVer) {
         case 4:
                 reset = rtl8125_test_phy_ocp_v4(tp);
@@ -16636,6 +16639,7 @@ rtl8125_test_phy_ocp(struct rtl8125_priv
         default:
                 goto unlock;
         }
+#endif
 
 unlock:
         r8125_spin_unlock(&tp->phy_lock, flags);
diff -Naurp r8125-9.016.01.orig/src/r8125_ptp.c r8125-9.016.01.el10_2/src/r8125_ptp.c
--- r8125-9.016.01.orig/src/r8125_ptp.c	2025-07-28 10:08:00.000000000 -0400
+++ r8125-9.016.01.el10_2/src/r8125_ptp.c	2026-05-22 10:59:33.579899026 -0400
@@ -1213,8 +1213,12 @@ void rtl8125_ptp_init(struct rtl8125_pri
         switch (tp->HwSuppPtpVer) {
         case 3:
                 tp->pps_enable = 0;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,13,0) && !(RHEL_MAJOR == 10 && RHEL_MINOR >= 2)
                 hrtimer_init(&tp->pps_timer, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
                 tp->pps_timer.function = rtl8125_phy_hrtimer_for_pps;
+#else
+                hrtimer_setup(&tp->pps_timer, rtl8125_phy_hrtimer_for_pps, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
+#endif
                 break;
         default:
                 break;
