diff -Naurp r8125-9.015.00.orig/src/r8125_ptp.c r8125-9.015.00/src/r8125_ptp.c
--- r8125-9.015.00.orig/src/r8125_ptp.c	2025-05-24 11:28:30.593109734 -0400
+++ r8125-9.015.00/src/r8125_ptp.c	2025-05-24 11:34:05.042077325 -0400
@@ -303,8 +303,13 @@ static int rtl8125_phc_enable(struct ptp
         }
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,11,0)
+int rtl8125_get_ts_info(struct net_device *netdev,
+                        struct kernel_ethtool_ts_info *info)
+#else
 int rtl8125_get_ts_info(struct net_device *netdev,
                         struct ethtool_ts_info *info)
+#endif
 {
         struct rtl8125_private *tp = netdev_priv(netdev);
 
diff -Naurp r8125-9.015.00.orig/src/r8125_ptp.h r8125-9.015.00/src/r8125_ptp.h
--- r8125-9.015.00.orig/src/r8125_ptp.h	2025-01-15 09:36:26.000000000 -0500
+++ r8125-9.015.00/src/r8125_ptp.h	2025-05-24 11:33:32.721597264 -0400
@@ -65,8 +65,13 @@ enum PTP_CMD_TYPE {
 struct rtl8125_private;
 struct RxDescV3;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,11,0)
+int rtl8125_get_ts_info(struct net_device *netdev,
+                        struct kernel_ethtool_ts_info *info);
+#else
 int rtl8125_get_ts_info(struct net_device *netdev,
                         struct ethtool_ts_info *info);
+#endif
 
 void rtl8125_ptp_reset(struct rtl8125_private *tp);
 void rtl8125_ptp_init(struct rtl8125_private *tp);
