diff -Naurp kmod-wl-6.30.223.271.ubuntu.el9/amd64/src/wl/sys/wl_cfg80211_hybrid.c kmod-wl-6.30.223.271.ubuntu.el9.wpa_supplicant/amd64/src/wl/sys/wl_cfg80211_hybrid.c
--- kmod-wl-6.30.223.271.ubuntu.el9/amd64/src/wl/sys/wl_cfg80211_hybrid.c	2026-09-04 02:03:21.044070753 -0400
+++ kmod-wl-6.30.223.271.ubuntu.el9.wpa_supplicant/amd64/src/wl/sys/wl_cfg80211_hybrid.c	2026-09-04 10:41:42.374288505 -0400
@@ -1976,6 +1976,13 @@ static s32 wl_alloc_wdev(struct device *
 	}
 	set_wiphy_dev(wdev->wiphy, dev);
 	wdev->wiphy->max_scan_ssids = WL_NUM_SCAN_MAX;
+
+	/* Set max_scan_ie_len to a random value in order to make wpa_supplicant
+	 * scans not to fail. The driver should ignore the extra passed IEs.
+	 * Value taken and fix inspired from:
+	 * https://patchwork.kernel.org/project/linux-wireless/patch/20211212221310.5453-1-merlijn@wizzup.org/
+	 */
+	wdev->wiphy->max_scan_ie_len = 512;
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 33)
 	wdev->wiphy->max_num_pmkids = WL_NUM_PMKIDS_MAX;
 #endif
