diff -Naurp kmod-wl-6.30.223.271.ubuntu.orig/amd64/src/wl/sys/wl_cfg80211_hybrid.c kmod-wl-6.30.223.271.ubuntu.el8/amd64/src/wl/sys/wl_cfg80211_hybrid.c
--- kmod-wl-6.30.223.271.ubuntu.orig/amd64/src/wl/sys/wl_cfg80211_hybrid.c	2026-09-05 10:39:32.566524711 -0400
+++ kmod-wl-6.30.223.271.ubuntu.el8/amd64/src/wl/sys/wl_cfg80211_hybrid.c	2026-09-05 10:39:41.096321925 -0400
@@ -120,7 +120,7 @@ static s32 wl_cfg80211_get_tx_power(stru
 static s32 wl_cfg80211_get_tx_power(struct wiphy *wiphy, s32 *dbm);
 #endif
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0) || RHEL_MAJOR >= 8
 static s32 wl_cfg80211_config_default_key(struct wiphy *wiphy,
            struct net_device *dev, int link_id, u8 key_idx, bool unicast,
            bool multicast);
@@ -141,7 +141,7 @@ static s32 wl_cfg80211_get_key(struct wi
            int link_id, u8 key_idx, bool pairwise, const u8 *mac_addr,
            void *cookie,
            void (*callback) (void *cookie, struct key_params *params));
-#elif LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0) || RHEL_MAJOR >= 8
 static s32 wl_cfg80211_add_key(struct wiphy *wiphy, struct net_device *dev,
            int link_id, u8 key_idx, bool pairwise, const u8 *mac_addr,
            struct key_params *params);
@@ -1223,7 +1223,7 @@ static s32 wl_cfg80211_get_tx_power(stru
 	return err;
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0) || RHEL_MAJOR >= 8
 static s32
 wl_cfg80211_config_default_key(struct wiphy *wiphy,
 	struct net_device *dev, int link_id, u8 key_idx, bool unicast,
@@ -1258,7 +1258,7 @@ static s32
 wl_cfg80211_add_key(struct wiphy *wiphy, struct wireless_dev *wdev,
                     int link_id, u8 key_idx, bool pairwise, const u8 *mac_addr,
                     struct key_params *params)
-#elif LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0) || RHEL_MAJOR >= 8
 static s32
 wl_cfg80211_add_key(struct wiphy *wiphy, struct net_device *dev,
                     int link_id, u8 key_idx, bool pairwise, const u8 *mac_addr,
@@ -1391,7 +1391,7 @@ wl_cfg80211_add_key(struct wiphy *wiphy,
 static s32
 wl_cfg80211_del_key(struct wiphy *wiphy, struct wireless_dev *wdev,
                     int link_id, u8 key_idx, bool pairwise, const u8 *mac_addr)
-#elif LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0) || RHEL_MAJOR >= 8
 static s32
 wl_cfg80211_del_key(struct wiphy *wiphy, struct net_device *dev,
                     int link_id, u8 key_idx, bool pairwise, const u8 *mac_addr)
@@ -1447,7 +1447,7 @@ wl_cfg80211_get_key(struct wiphy *wiphy,
                     int link_id, u8 key_idx, bool pairwise, const u8 *mac_addr,
                     void *cookie,
                     void (*callback) (void *cookie, struct key_params * params))
-#elif LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0) || RHEL_MAJOR >= 8
 static s32
 wl_cfg80211_get_key(struct wiphy *wiphy, struct net_device *dev,
                     int link_id, u8 key_idx, bool pairwise, const u8 *mac_addr,
@@ -2496,7 +2496,7 @@ wl_bss_roaming_done(struct wl_cfg80211_p
 	struct wl_cfg80211_connect_info *conn_info = wl_to_conn(wl);
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 12, 0)
 	struct cfg80211_roam_info roam_info = {
-#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 0, 0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 0, 0) && RHEL_MAJOR < 8
 		.bssid = wl->profile->bssid,
 #else
 		.links[0].bssid = wl->profile->bssid,
