diff -Naurp arcmsr-1.40.00.05.orig/arcmsr_hba.c arcmsr-1.40.00.05/arcmsr_hba.c
--- arcmsr-1.40.00.05.orig/arcmsr_hba.c	2018-11-21 08:22:14.000000000 +0000
+++ arcmsr-1.40.00.05/arcmsr_hba.c	2021-05-18 20:44:37.895200704 +0100
@@ -350,16 +350,11 @@ static irqreturn_t arcmsr_do_interrupt(i
 static int arcmsr_bios_param(struct scsi_device *sdev,
 		struct block_device *bdev, sector_t capacity, int *geom)
 {
-	int ret, heads, sectors, cylinders, total_capacity;
-	unsigned char *buffer;/* return copy of block device's partition table */
+	int heads, sectors, cylinders, total_capacity;
+
+	if (scsi_partsize(bdev, capacity, geom))
+		return 0;
 
-	buffer = scsi_bios_ptable(bdev);
-	if (buffer) {
-		ret = scsi_partsize(buffer, capacity, &geom[2], &geom[0], &geom[1]);
-		kfree(buffer);
-		if (ret != -1)
-			return ret;
-	}
 	total_capacity = capacity;
 	heads = 64;
 	sectors = 32;
