diff -Npru a/src/kcompat.h b/src/kcompat.h
--- a/src/kcompat.h	2016-05-10 06:10:28.000000000 -0400
+++ b/src/kcompat.h	2016-12-29 13:31:09.946969744 -0500
@@ -4788,7 +4788,7 @@ static inline void __kc_timecounter_adjt
 #define HAVE_NDO_SET_VF_RSS_QUERY_EN
 #endif /* 4,1,0 */
 
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,1,9))
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,1,9)) && (RHEL_MAJOR != 7)
 #if (!(SLE_VERSION_CODE && SLE_VERSION_CODE >= SLE_VERSION(12,1,0)))
 static inline bool page_is_pfmemalloc(struct page __maybe_unused * page)
 {
