--- src/rxkad/Makefile.in.orig	2013-02-21 07:12:06.000000000 -0500
+++ src/rxkad/Makefile.in	2013-03-04 15:05:58.594113861 -0500
@@ -77,10 +77,10 @@
 	${COMPILE_ET} -p ${srcdir} rxkad_errs -h rxkad
 
 ticket.o: ticket.c lifetimes.h ${INCLS}
-	$(CC) $(CFLAGS) -c ${srcdir}/ticket.c
+	$(CCOBJ) $(CFLAGS) -c ${srcdir}/ticket.c
 
 ticket5.o: ticket5.c v5gen.c v5der.c v5gen-rewrite.h ${INCLS}
-	$(CC) -I${srcdir} $(CFLAGS) -c ${srcdir}/ticket5.c
+	$(CCOBJ) -I${srcdir} $(CFLAGS) -c ${srcdir}/ticket5.c
 
 crc.o: crc.c ${INCLS}
 
