diff -up ecryptfs-utils-108/src/include/ecryptfs.h.syslog ecryptfs-utils-108/src/include/ecryptfs.h
--- ecryptfs-utils-108/src/include/ecryptfs.h.syslog	2015-08-11 14:44:02.641814020 +0200
+++ ecryptfs-utils-108/src/include/ecryptfs.h	2015-08-11 14:44:02.643814015 +0200
@@ -137,7 +137,7 @@
 #define ECRYPTFS_TAG_67_PACKET 0x43
 
 #define ecryptfs_syslog(type, fmt, arg...) \
-	syslog(type, "%s: " fmt, __FUNCTION__, ## arg);
+	syslog(type, "ecryptfs: %s: " fmt, __FUNCTION__, ## arg);
 
 #define ECRYPTFS_MAX_NUM_CIPHERS 64
 #define ECRYPTFS_ECHO_ON 1
