diff -Naurp drivers-leds.el9_2/leds-gpio.c drivers-leds.el9_3/leds-gpio.c
--- drivers-leds.el9_2/leds-gpio.c	2022-03-24 19:27:30.000000000 -0400
+++ drivers-leds.el9_3/leds-gpio.c	2023-12-12 22:14:03.806511248 -0500
@@ -151,9 +151,9 @@ static struct gpio_leds_priv *gpio_leds_
 		 * will be updated after LED class device is registered,
 		 * Only then the final LED name is known.
 		 */
-		led.gpiod = devm_fwnode_get_gpiod_from_child(dev, NULL, child,
-							     GPIOD_ASIS,
-							     NULL);
+		led.gpiod = devm_fwnode_gpiod_get(dev, child, NULL, 
+						  GPIOD_ASIS,
+						  NULL);
 		if (IS_ERR(led.gpiod)) {
 			fwnode_handle_put(child);
 			return ERR_CAST(led.gpiod);
