--- qla_os.c.orig	2023-04-05 17:44:47.000000000 +0100
+++ qla_os.c	2023-07-13 11:12:55.556681165 +0100
@@ -2744,7 +2744,7 @@ static void qla2x00_iocb_work_fn(struct
 	spin_unlock_irqrestore(&vha->work_lock, flags);
 }
 
-#ifdef CONFIG_RHEL_DIFFERENCES
+#ifdef CONFIG_ELREPO_DIFFERENCES
 static const struct pci_device_id rh_deprecated_pci_table[] = {
 	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2532) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2031) },
@@ -2780,7 +2780,7 @@ qla2x00_probe_one(struct pci_dev *pdev,
 	struct rsp_que *rsp = NULL;
 	int i;
 
-#ifdef CONFIG_RHEL_DIFFERENCES
+#ifdef CONFIG_ELREPO_DIFFERENCES
 	if (pci_hw_disabled(rh_disabled_pci_table, pdev))
 		return -ENODEV;
 
