--- qla_os.c.orig	2022-04-16 01:03:46.000000000 +0100
+++ qla_os.c	2022-05-11 20:58:49.372053818 +0100
@@ -2689,23 +2689,6 @@ static void qla2x00_iocb_work_fn(struct
 	spin_unlock_irqrestore(&vha->work_lock, flags);
 }
 
-#ifdef CONFIG_RHEL_DIFFERENCES
-static const struct pci_device_id rh_deprecated_pci_table[] = {
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2532) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP2031) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_QLOGIC, PCI_DEVICE_ID_QLOGIC_ISP8031) },
-	{ 0 }
-};
-
-static const struct pci_device_id rh_unmaintained_pci_table[] = {
-	{ 0 }
-};
-
-static const struct pci_device_id rh_disabled_pci_table[] = {
-	{ 0 }
-};
-#endif
-
 /*
  * PCI driver interface
  */
@@ -2725,14 +2708,6 @@ qla2x00_probe_one(struct pci_dev *pdev,
 	struct rsp_que *rsp = NULL;
 	int i;
 
-#ifdef CONFIG_RHEL_DIFFERENCES
-	if (pci_hw_disabled(rh_disabled_pci_table, pdev))
-		return -ENODEV;
-
-	pci_hw_deprecated(rh_deprecated_pci_table, pdev);
-	pci_hw_unmaintained(rh_unmaintained_pci_table, pdev);
-#endif
-
 	bars = pci_select_bars(pdev, IORESOURCE_MEM | IORESOURCE_IO);
 	sht = &qla2xxx_driver_template;
 	if (pdev->device == PCI_DEVICE_ID_QLOGIC_ISP2532 ||
