--- megaraid_sas_base.c.orig	2022-04-14 16:51:12.000000000 +0100
+++ megaraid_sas_base.c	2022-05-17 15:13:25.642073713 +0100
@@ -144,7 +144,7 @@ static void megasas_get_pd_info(struct m
 static void
 megasas_set_ld_removed_by_fw(struct megasas_instance *instance);
 
-#ifdef CONFIG_RHEL_DIFFERENCES
+#ifdef CONFIG_ELREPO_DIFFERENCES
 static const struct pci_device_id rh_deprecated_pci_table[] = {
 
 	{0}     /* Terminating entry */
@@ -7503,7 +7503,7 @@ static int megasas_probe_one(struct pci_
 	struct megasas_instance *instance;
 	u16 control = 0;
 
-#ifdef CONFIG_RHEL_DIFFERENCES
+#ifdef CONFIG_ELREPO_DIFFERENCES
 	if (pci_hw_disabled(rh_disabled_pci_table, pdev))
 		return -ENODEV;
 
