diff -Naurp jme-1.0.8.orig/jme.c jme-1.0.8/jme.c
--- jme-1.0.8.orig/jme.c	2021-11-21 12:49:09.000000000 +0000
+++ jme-1.0.8/jme.c	2023-05-09 17:14:03.707389880 +0100
@@ -3022,7 +3022,7 @@ jme_init_one(struct pci_dev *pdev,
 		jwrite32(jme, JME_APMC, apmc);
 	}
 
-	NETIF_NAPI_SET(netdev, &jme->napi, jme_poll, NAPI_POLL_WEIGHT)
+	NETIF_NAPI_SET(netdev, &jme->napi, jme_poll)
 
 	spin_lock_init(&jme->phy_lock);
 	spin_lock_init(&jme->macaddr_lock);
diff -Naurp jme-1.0.8.orig/jme.h jme-1.0.8/jme.h
--- jme-1.0.8.orig/jme.h	2021-11-21 12:49:09.000000000 +0000
+++ jme-1.0.8/jme.h	2023-05-09 17:14:27.314129344 +0100
@@ -379,8 +379,8 @@ struct jme_ring {
 #define DECLARE_NET_DEVICE_STATS
 
 #define DECLARE_NAPI_STRUCT struct napi_struct napi;
-#define NETIF_NAPI_SET(dev, napis, pollfn, q) \
-	netif_napi_add(dev, napis, pollfn, q);
+#define NETIF_NAPI_SET(dev, napis, pollfn) \
+	netif_napi_add(dev, napis, pollfn);
 #define JME_NAPI_HOLDER(holder) struct napi_struct *holder
 #define JME_NAPI_WEIGHT(w) int w
 #define JME_NAPI_WEIGHT_VAL(w) w
