diff -Naurp a/sky2.c b/sky2.c
--- a/sky2.c	2020-11-05 20:35:50.000000000 +0000
+++ b/sky2.c	2022-11-16 10:33:06.865334751 +0000
@@ -4156,7 +4156,9 @@ static unsigned long roundup_ring_size(u
 }
 
 static void sky2_get_ringparam(struct net_device *dev,
-			       struct ethtool_ringparam *ering)
+			       struct ethtool_ringparam *ering,
+			       struct kernel_ethtool_ringparam *kernel_ering,
+			       struct netlink_ext_ack *extack)
 {
 	struct sky2_port *sky2 = netdev_priv(dev);
 
@@ -4168,7 +4170,9 @@ static void sky2_get_ringparam(struct ne
 }
 
 static int sky2_set_ringparam(struct net_device *dev,
-			      struct ethtool_ringparam *ering)
+			      struct ethtool_ringparam *ering,
+			      struct kernel_ethtool_ringparam *kernel_ering,
+			      struct netlink_ext_ack *extack)
 {
 	struct sky2_port *sky2 = netdev_priv(dev);
 
