diff -up VirtualGL-2.3.2/server/vglrun.redhatpathsmultilibfix VirtualGL-2.3.2/server/vglrun
--- VirtualGL-2.3.2/server/vglrun.redhatpathsmultilibfix	2013-04-20 05:54:45.520000000 -0400
+++ VirtualGL-2.3.2/server/vglrun	2013-04-20 05:55:26.107000022 -0400
@@ -220,17 +220,17 @@ SunOS)
 
 if [ -z "$FAKERLIB" ]; then
 FAKERLIB=librrfaker.so
-if [ -f /opt/VirtualGL/lib/64/librrfaker.so ]
+if [ -f /usr/lib64/VirtualGL/librrfaker.so ]
 then
-	FAKERLIB=/opt/VirtualGL/lib/64/librrfaker.so
+	FAKERLIB=/usr/lib64/VirtualGL/librrfaker.so
 fi
 fi
 export FAKERLIB
 if [ -z "$FAKERLIB32" ]; then
 FAKERLIB32=librrfaker.so
-if [ -f /opt/VirtualGL/lib/librrfaker.so ]
+if [ -f /usr/lib/VirtualGL/librrfaker.so ]
 then
-	FAKERLIB32=/opt/VirtualGL/lib/librrfaker.so
+	FAKERLIB32=/usr/lib/VirtualGL/librrfaker.so
 fi
 fi
 export FAKERLIB32
@@ -241,17 +241,17 @@ if [ $__VGL_64 -eq 0 ]; then
 		LD_PRELOAD_32=$FAKERLIB32:$LD_PRELOAD_32
 	fi
 	if [ $__VGL_DL -eq 1 ]; then
-		if [ -f /opt/VirtualGL/lib/libdlfaker.so ]
+		if [ -f /usr/lib/VirtualGL/libdlfaker.so ]
 		then
-			LD_PRELOAD_32=/opt/VirtualGL/lib/libdlfaker.so:$LD_PRELOAD_32
+			LD_PRELOAD_32=/usr/lib/VirtualGL/libdlfaker.so:$LD_PRELOAD_32
 		else
 			LD_PRELOAD_32=libdlfaker.so:$LD_PRELOAD_32
 		fi
 	fi
 	if [ $__VGL_GE -eq 1 ]; then
-		if [ -f /opt/VirtualGL/lib/libgefaker.so ]
+		if [ -f /usr/lib/VirtualGL/libgefaker.so ]
 		then
-			LD_PRELOAD_32=/opt/VirtualGL/lib/libgefaker.so:$LD_PRELOAD_32
+			LD_PRELOAD_32=/usr/lib/VirtualGL/libgefaker.so:$LD_PRELOAD_32
 		else
 			LD_PRELOAD_32=libgefaker.so:$LD_PRELOAD_32
 		fi
@@ -265,17 +265,17 @@ if [ $__VGL_32 -eq 0 ]; then
 		LD_PRELOAD_64=$FAKERLIB:$LD_PRELOAD_64
 	fi
 	if [ $__VGL_DL -eq 1 ]; then
-		if [ -f /opt/VirtualGL/lib/64/libdlfaker.so ]
+		if [ -f /usr/lib64/VirtualGL/libdlfaker.so ]
 		then
-			LD_PRELOAD_64=/opt/VirtualGL/lib/64/libdlfaker.so:$LD_PRELOAD_64
+			LD_PRELOAD_64=/usr/lib64/VirtualGL/libdlfaker.so:$LD_PRELOAD_64
 		else
 			LD_PRELOAD_64=libdlfaker.so:$LD_PRELOAD_64
 		fi
 	fi
 	if [ $__VGL_GE -eq 1 ]; then
-		if [ -f /opt/VirtualGL/lib/64/libgefaker.so ]
+		if [ -f /usr/lib64/VirtualGL/libgefaker.so ]
 		then
-			LD_PRELOAD_64=/opt/VirtualGL/lib/64/libgefaker.so:$LD_PRELOAD_64
+			LD_PRELOAD_64=/usr/lib64/VirtualGL/libgefaker.so:$LD_PRELOAD_64
 		else
 			LD_PRELOAD_64=libgefaker.so:$LD_PRELOAD_64
 		fi
@@ -288,16 +288,18 @@ exec ${1+"$@"}
 
 *)
 
+LD_LIBRARY_PATH=/usr/lib64/VirtualGL:/usr/lib/VirtualGL:$LD_LIBRARY_PATH
+
 if [ -z "$LD_PRELOAD" ]; then
-	LD_PRELOAD=librrfaker.so
+	LD_PRELOAD=/usr/\$LIB/VirtualGL/librrfaker.so
 else
 	LD_PRELOAD=librrfaker.so:$LD_PRELOAD
 fi
 if [ $__VGL_DL -eq 1 ]; then
-	LD_PRELOAD=libdlfaker.so:$LD_PRELOAD
+	LD_PRELOAD=/usr/\$LIB/VirtualGL/libdlfaker.so:$LD_PRELOAD
 fi
 if [ $__VGL_GE -eq 1 ]; then
-	LD_PRELOAD=libgefaker.so:$LD_PRELOAD
+	LD_PRELOAD=/usr/\$LIB/VirtualGL/libgefaker.so:$LD_PRELOAD
 fi
 export LD_PRELOAD
 exec ${1+"$@"}
