--- mptsas.c.orig	2022-04-14 16:51:12.000000000 +0100
+++ mptsas.c	2022-05-18 17:44:07.239078295 +0100
@@ -5163,7 +5163,7 @@ mptsas_probe(struct pci_dev *pdev, const
 	int			error=0;
 	int			r;
 
-#ifdef CONFIG_RHEL_DIFFERENCES
+#ifdef CONFIG_ELREPO_DIFFERENCES
 	mark_driver_unmaintained(MYNAM);
 #endif
 
@@ -5385,7 +5385,7 @@ static void mptsas_remove(struct pci_dev
 }
 
 static struct pci_device_id mptsas_pci_table[] = {
-#ifdef CONFIG_RHEL_DIFFERENCES
+#ifdef CONFIG_ELREPO_DIFFERENCES
 	{ PCI_VENDOR_ID_LSI_LOGIC, MPI_MANUFACTPAGE_DEVID_SAS1068,
 		PCI_VENDOR_ID_VMWARE, PCI_ANY_ID },
 #else
