diff -Naurp r8127-11.014.00.orig/src/r8127_ptp.c r8127-11.014.00/src/r8127_ptp.c
--- r8127-11.014.00.orig/src/r8127_ptp.c	2025-03-28 10:04:19.000000000 -0400
+++ r8127-11.014.00/src/r8127_ptp.c	2025-05-24 11:56:17.015854217 -0400
@@ -356,8 +356,13 @@ static void rtl8127_ptp_enable_config(st
         rtl8127_set_eth_phy_ocp_bit(tp, 0xA640, BIT_15);
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,11,0)
+int rtl8127_get_ts_info(struct net_device *netdev,
+                        struct kernel_ethtool_ts_info *info)
+#else
 int rtl8127_get_ts_info(struct net_device *netdev,
                         struct ethtool_ts_info *info)
+#endif
 {
         struct rtl8127_private *tp = netdev_priv(netdev);
 
diff -Naurp r8127-11.014.00.orig/src/r8127_ptp.h r8127-11.014.00/src/r8127_ptp.h
--- r8127-11.014.00.orig/src/r8127_ptp.h	2025-03-28 10:04:19.000000000 -0400
+++ r8127-11.014.00/src/r8127_ptp.h	2025-05-24 11:55:53.536505598 -0400
@@ -185,8 +185,13 @@ static inline u8 ptp_get_msgtype(const s
 struct rtl8127_private;
 struct RxDescV3;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,11,0)
+int rtl8127_get_ts_info(struct net_device *netdev,
+                        struct kernel_ethtool_ts_info *info);
+#else
 int rtl8127_get_ts_info(struct net_device *netdev,
                         struct ethtool_ts_info *info);
+#endif
 
 void rtl8127_ptp_reset(struct rtl8127_private *tp);
 void rtl8127_ptp_init(struct rtl8127_private *tp);
