diff -Naurp kmod-nvidia-390xx-390.157.stock/nvidiapkg/kernel/nvidia-drm/nvidia-drm-linux.c kmod-nvidia-390xx-390.157/nvidiapkg/kernel/nvidia-drm/nvidia-drm-linux.c
--- kmod-nvidia-390xx-390.157.stock/nvidiapkg/kernel/nvidia-drm/nvidia-drm-linux.c	2022-10-12 05:30:31.000000000 -0400
+++ kmod-nvidia-390xx-390.157/nvidiapkg/kernel/nvidia-drm/nvidia-drm-linux.c	2024-09-02 09:58:01.204335323 -0400
@@ -41,8 +41,8 @@
 
 MODULE_PARM_DESC(
     modeset,
-    "Enable atomic kernel modesetting (1 = enable, 0 = disable (default))");
-bool nv_drm_modeset_module_param = false;
+    "Enable atomic kernel modesetting (1 = enable (default), 0 = disable)");
+bool nv_drm_modeset_module_param = true;
 module_param_named(modeset, nv_drm_modeset_module_param, bool, 0400);
 
 void *nv_drm_calloc(size_t nmemb, size_t size)
