diff -Naurp r8126-10.017.00.orig/src/r8126_ptp.c r8126-10.017.00/src/r8126_ptp.c
--- r8126-10.017.00.orig/src/r8126_ptp.c	2025-12-08 08:52:06.000000000 -0500
+++ r8126-10.017.00/src/r8126_ptp.c	2026-07-03 11:06:08.867637462 -0400
@@ -748,7 +748,7 @@ void rtl8126_ptp_init(struct rtl8126_pri
 
         /* init a hrtimer for pps */
         tp->pps_enable = 0;
-#if LINUX_VERSION_CODE < KERNEL_VERSION(6,13,0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,13,0) && !(RHEL_MAJOR == 10 && RHEL_MINOR >= 2)
         hrtimer_init(&tp->pps_timer, CLOCK_MONOTONIC, HRTIMER_MODE_REL);
         tp->pps_timer.function = rtl8126_hrtimer_for_pps;
 #else
