diff -Naurp r8126-10.015.00.orig/src/r8126_ptp.c r8126-10.015.00/src/r8126_ptp.c
--- r8126-10.015.00.orig/src/r8126_ptp.c	2025-01-15 09:48:21.000000000 -0500
+++ r8126-10.015.00/src/r8126_ptp.c	2025-05-24 11:47:19.843878601 -0400
@@ -356,8 +356,13 @@ static void rtl8126_ptp_enable_config(st
         rtl8126_set_eth_phy_ocp_bit(tp, 0xA640, BIT_15);
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,11,0)
+int rtl8126_get_ts_info(struct net_device *netdev,
+                        struct kernel_ethtool_ts_info *info)
+#else
 int rtl8126_get_ts_info(struct net_device *netdev,
                         struct ethtool_ts_info *info)
+#endif
 {
         struct rtl8126_private *tp = netdev_priv(netdev);
 
diff -Naurp r8126-10.015.00.orig/src/r8126_ptp.h r8126-10.015.00/src/r8126_ptp.h
--- r8126-10.015.00.orig/src/r8126_ptp.h	2025-01-15 09:48:21.000000000 -0500
+++ r8126-10.015.00/src/r8126_ptp.h	2025-05-24 11:46:49.331425570 -0400
@@ -185,8 +185,13 @@ static inline u8 ptp_get_msgtype(const s
 struct rtl8126_private;
 struct RxDescV3;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,11,0)
+int rtl8126_get_ts_info(struct net_device *netdev,
+                        struct kernel_ethtool_ts_info *info);
+#else
 int rtl8126_get_ts_info(struct net_device *netdev,
                         struct ethtool_ts_info *info);
+#endif
 
 void rtl8126_ptp_reset(struct rtl8126_private *tp);
 void rtl8126_ptp_init(struct rtl8126_private *tp);
