diff -Naurp ib_qib-1.11.orig/qib_user_pages.c ib_qib-1.11/qib_user_pages.c
--- ib_qib-1.11.orig/qib_user_pages.c	2021-04-29 12:35:54.000000000 +0100
+++ ib_qib-1.11/qib_user_pages.c	2021-05-25 10:36:14.675505159 +0100
@@ -40,7 +40,7 @@
 static void __qib_release_user_pages(struct page **p, size_t num_pages,
 				     int dirty)
 {
-	put_user_pages_dirty_lock(p, num_pages, dirty);
+	unpin_user_pages_dirty_lock(p, num_pages, dirty);
 }
 
 /*
diff -Naurp ib_qib-1.11.orig/qib_user_sdma.c ib_qib-1.11/qib_user_sdma.c
--- ib_qib-1.11.orig/qib_user_sdma.c	2021-04-29 12:35:54.000000000 +0100
+++ ib_qib-1.11/qib_user_sdma.c	2021-05-25 10:44:16.766168974 +0100
@@ -320,7 +320,7 @@ static int qib_user_sdma_page_to_frags(c
 		 * the caller can ignore this page.
 		 */
 		if (put) {
-			put_user_page(page);
+			unpin_user_page(page);
 		} else {
 			/* coalesce case */
 			kunmap(page);
@@ -634,7 +634,7 @@ static void qib_user_sdma_free_pkt_frag(
 			kunmap(pkt->addr[i].page);
 
 		if (pkt->addr[i].put_page)
-			put_user_page(pkt->addr[i].page);
+			unpin_user_page(pkt->addr[i].page);
 		else
 			__free_page(pkt->addr[i].page);
 	} else if (pkt->addr[i].kvaddr) {
@@ -709,7 +709,7 @@ static int qib_user_sdma_pin_pages(const
 	/* if error, return all pages not managed by pkt */
 free_pages:
 	while (i < j)
-		put_user_page(pages[i++]);
+		unpin_user_page(pages[i++]);
 
 done:
 	return ret;
