diff -Naurp qlge-1.00.00.35.orig/qlge_main.c qlge-1.00.00.35.el9_8/qlge_main.c
--- qlge-1.00.00.35.orig/qlge_main.c	2023-10-29 22:31:08.000000000 -0400
+++ qlge-1.00.00.35.el9_8/qlge_main.c	2026-05-23 23:16:55.162162506 -0400
@@ -4519,7 +4519,7 @@ static const struct net_device_ops qlge_
 
 static void qlge_timer(struct timer_list *t)
 {
-	struct qlge_adapter *qdev = from_timer(qdev, t, timer);
+	struct qlge_adapter *qdev = timer_container_of(qdev, t, timer);
 	u32 var = 0;
 
 	var = qlge_read32(qdev, STS);
