diff -Naurp a/osm/linux/os_linux.c b/osm/linux/os_linux.c
--- a/osm/linux/os_linux.c	2021-10-26 08:22:36.000000000 +0100
+++ b/osm/linux/os_linux.c	2023-11-11 12:54:56.904363034 +0000
@@ -182,7 +182,7 @@ void freelist_put_dma(struct freelist *l
 
 BUS_ADDRESS get_dmapool_phy_addr(void *osext, void * dmapool_virt_addr)
 {
-	return (BUS_ADDRESS)virt_to_bus(dmapool_virt_addr);
+	return (BUS_ADDRESS)virt_to_phys(dmapool_virt_addr);
 }
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,18)) && defined(CONFIG_HIGHMEM)
