diff -Naurp rdma-core-48.0.orig/CMakeLists.txt rdma-core-48.0/CMakeLists.txt
--- rdma-core-48.0.orig/CMakeLists.txt	2024-09-09 09:17:13.154339853 -0400
+++ rdma-core-48.0/CMakeLists.txt	2024-09-09 09:17:08.557437722 -0400
@@ -729,7 +729,7 @@ add_subdirectory(providers/mlx4)
 add_subdirectory(providers/mlx4/man)
 add_subdirectory(providers/mlx5)
 add_subdirectory(providers/mlx5/man)
-#add_subdirectory(providers/mthca)
+add_subdirectory(providers/mthca)
 #add_subdirectory(providers/ocrdma)
 add_subdirectory(providers/qedr)
 add_subdirectory(providers/vmw_pvrdma)
