diff -Naurp r8126-10.013.00.elrepo/src/r8126.h r8126-10.013.00/src/r8126.h
--- r8126-10.013.00.elrepo/src/r8126.h	2024-01-22 08:11:04.000000000 -0500
+++ r8126-10.013.00/src/r8126.h	2024-04-12 11:52:09.201449959 -0400
@@ -51,7 +51,7 @@
 #define fallthrough
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(4,3,0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,3,0) && RHEL_MAJOR != 7
 static inline
 ssize_t strscpy(char *dest, const char *src, size_t count)
 {
@@ -79,7 +79,7 @@ ssize_t strscpy(char *dest, const char *
 }
 #endif
 
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,6,0))
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4,6,0)) && RHEL_MAJOR != 7
 static inline unsigned char *skb_checksum_start(const struct sk_buff *skb)
 {
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,22))
@@ -151,7 +151,7 @@ static inline bool dev_page_is_reusable(
 #define device_set_wakeup_enable(dev, val)	do {} while (0)
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(3,14,0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,14,0) && RHEL_MAJOR != 7
 static inline void ether_addr_copy(u8 *dst, const u8 *src)
 {
         u16 *a = (u16 *)dst;
diff -Naurp r8126-10.013.00.elrepo/src/r8126_n.c r8126-10.013.00/src/r8126_n.c
--- r8126-10.013.00.elrepo/src/r8126_n.c	2024-01-22 08:07:24.000000000 -0500
+++ r8126-10.013.00/src/r8126_n.c	2024-04-12 11:54:28.902563537 -0400
@@ -13516,7 +13516,11 @@ static const struct net_device_ops rtl81
 #endif //LINUX_VERSION_CODE >= KERNEL_VERSION(5,11,0)
         .ndo_start_xmit     = rtl8126_start_xmit,
         .ndo_tx_timeout     = rtl8126_tx_timeout,
+#if RHEL_MAJOR == 7
+        .ndo_change_mtu_rh74= rtl8126_change_mtu,
+#else
         .ndo_change_mtu     = rtl8126_change_mtu,
+#endif //RHEL_MAJOR == 7
         .ndo_set_mac_address    = rtl8126_set_mac_address,
 #if LINUX_VERSION_CODE < KERNEL_VERSION(5,15,0)
         .ndo_do_ioctl       = rtl8126_do_ioctl,
@@ -16831,9 +16835,9 @@ rtl8126_rx_interrupt(struct net_device *
                 if (rtl8126_rx_vlan_skb(tp, desc, skb) < 0)
                         rtl8126_rx_skb(tp, skb, ring_index);
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0) && RHEL_MAJOR != 7
                 dev->last_rx = jiffies;
-#endif //LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0)
+#endif //LINUX_VERSION_CODE < KERNEL_VERSION(4,11,0) && RHEL_MAJOR != 7
                 total_rx_packets++;
 
 #ifdef ENABLE_PAGE_REUSE
