diff -Naurp r8127-11.014.00.orig/src/r8127.h r8127-11.014.00/src/r8127.h
--- r8127-11.014.00.orig/src/r8127.h	2025-03-28 10:04:19.000000000 -0400
+++ r8127-11.014.00/src/r8127.h	2025-04-29 12:55:30.551400854 -0400
@@ -37,6 +37,7 @@
 
 //#include <linux/pci.h>
 #include <linux/ethtool.h>
+#include <linux/mdio.h>
 #include <linux/interrupt.h>
 #include <linux/version.h>
 #include "r8127_dash.h"
@@ -517,7 +518,7 @@ do { \
 #define  MDIO_EEE_5GT  0x0002
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(6,9,0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,9,0) && RHEL_MAJOR != 9
 #define ethtool_keee ethtool_eee
 #define rtl8127_ethtool_adv_to_mmd_eee_adv_cap1_t ethtool_adv_to_mmd_eee_adv_t
 static inline u32 rtl8127_ethtool_adv_to_mmd_eee_adv_cap2_t(u32 adv)
@@ -530,6 +531,27 @@ static inline u32 rtl8127_ethtool_adv_to
         return result;
 }
 #else
+#if RHEL_MAJOR == 9
+static inline u32 linkmode_to_mii_eee_cap2_t(unsigned long *adv)
+{
+	u32 result = 0;
+
+	if (linkmode_test_bit(ETHTOOL_LINK_MODE_2500baseT_Full_BIT, adv))
+		result |= MDIO_EEE_2_5GT;
+	if (linkmode_test_bit(ETHTOOL_LINK_MODE_5000baseT_Full_BIT, adv))
+		result |= MDIO_EEE_5GT;
+
+	return result;
+}
+
+static inline void mii_eee_cap2_mod_linkmode_sup_t(unsigned long *adv, u32 val)
+{
+	linkmode_mod_bit(ETHTOOL_LINK_MODE_2500baseT_Full_BIT,
+			 adv, val & MDIO_EEE_2_5GT);
+	linkmode_mod_bit(ETHTOOL_LINK_MODE_5000baseT_Full_BIT,
+			 adv, val & MDIO_EEE_5GT);
+}
+#endif
 #define rtl8127_ethtool_adv_to_mmd_eee_adv_cap1_t linkmode_to_mii_eee_cap1_t
 #define rtl8127_ethtool_adv_to_mmd_eee_adv_cap2_t linkmode_to_mii_eee_cap2_t
 #endif /* LINUX_VERSION_CODE < KERNEL_VERSION(6,9,0) */
@@ -933,7 +955,7 @@ typedef int *napi_budget;
 typedef struct napi_struct *napi_ptr;
 typedef int napi_budget;
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,1,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,1,0) || RHEL_MAJOR == 9
 #define RTL_NAPI_CONFIG(ndev, priv, function, weight)   netif_napi_add_weight(ndev, &priv->napi, function, weight)
 #else
 #define RTL_NAPI_CONFIG(ndev, priv, function, weight)   netif_napi_add(ndev, &priv->napi, function, weight)
diff -Naurp r8127-11.014.00.orig/src/r8127_n.c r8127-11.014.00/src/r8127_n.c
--- r8127-11.014.00.orig/src/r8127_n.c	2025-03-28 10:04:19.000000000 -0400
+++ r8127-11.014.00/src/r8127_n.c	2025-04-29 12:55:30.552400927 -0400
@@ -2736,7 +2736,7 @@ static void rtl8127_proc_module_init(voi
 static int rtl8127_proc_open(struct inode *inode, struct file *file)
 {
         struct net_device *dev = proc_get_parent_data(inode);
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,17,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,17,0) || RHEL_MAJOR == 9
         int (*show)(struct seq_file *, void *) = pde_data(inode);
 #else
         int (*show)(struct seq_file *, void *) = PDE_DATA(inode);
@@ -6472,7 +6472,7 @@ rtl8127_set_ring_size(struct rtl8127_pri
 }
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,17,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,17,0) || RHEL_MAJOR == 9
 static void rtl8127_get_ringparam(struct net_device *dev,
                                   struct ethtool_ringparam *ring,
                                   struct kernel_ethtool_ringparam *kernel_ring,
@@ -6490,7 +6490,7 @@ static void rtl8127_get_ringparam(struct
         ring->tx_pending = tp->tx_ring[0].num_tx_desc;
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,17,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,17,0) || RHEL_MAJOR == 9
 static int rtl8127_set_ringparam(struct net_device *dev,
                                  struct ethtool_ringparam *ring,
                                  struct kernel_ethtool_ringparam *kernel_ring,
@@ -6867,7 +6867,7 @@ rtl8127_device_lpi_t_to_ethtool_lpi_t(st
         return to_us;
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,9,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,9,0) || RHEL_MAJOR == 9
 static void
 rtl8127_adv_to_linkmode(unsigned long *mode, u64 adv)
 {
@@ -12348,7 +12348,7 @@ rtl8127_init_software_variable(struct ne
                 struct ethtool_keee *eee = &tp->eee;
 
                 eee->eee_enabled = eee_enable;
-#if LINUX_VERSION_CODE < KERNEL_VERSION(6,9,0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,9,0) && RHEL_MAJOR != 9
                 eee->supported  = SUPPORTED_100baseT_Full |
                                   SUPPORTED_1000baseT_Full |
                                   SUPPORTED_2500baseX_Full;
@@ -14285,7 +14285,7 @@ rtl8127_init_one(struct pci_dev *pdev,
                 dev->hw_features |= NETIF_F_IPV6_CSUM | NETIF_F_TSO6;
                 dev->features |= NETIF_F_IPV6_CSUM;
                 dev->features |= NETIF_F_TSO6;
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,19,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5,19,0) || RHEL_MAJOR == 9
                 netif_set_tso_max_size(dev, LSO_64K);
                 netif_set_tso_max_segs(dev, NIC_MAX_PHYS_BUF_COUNT_LSO2);
 #else //LINUX_VERSION_CODE >= KERNEL_VERSION(5,19,0)
diff -Naurp r8127-11.014.00.orig/src/r8127_ptp.c r8127-11.014.00/src/r8127_ptp.c
--- r8127-11.014.00.orig/src/r8127_ptp.c	2025-03-28 10:04:19.000000000 -0400
+++ r8127-11.014.00/src/r8127_ptp.c	2025-04-29 12:55:30.553401001 -0400
@@ -222,7 +222,7 @@ static int _rtl8127_phc_adjfreq(struct p
         return 0;
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,2,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,2,0) || RHEL_MAJOR == 9
 static int rtl8127_ptp_adjfine(struct ptp_clock_info *ptp, long scaled_ppm)
 {
         s32 ppb = scaled_ppm_to_ppb(scaled_ppm);
@@ -399,7 +399,7 @@ static const struct ptp_clock_info rtl_p
         .n_per_out  = 0,
         .n_pins     = 0,
         .pps        = 1,
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,2,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,2,0) || RHEL_MAJOR == 9
         .adjfine   = rtl8127_ptp_adjfine,
 #else
         .adjfreq    = rtl8127_phc_adjfreq,
diff -Naurp r8127-11.014.00.orig/src/r8127_rss.c r8127-11.014.00/src/r8127_rss.c
--- r8127-11.014.00.orig/src/r8127_rss.c	2025-03-28 10:04:19.000000000 -0400
+++ r8127-11.014.00/src/r8127_rss.c	2025-04-29 12:55:30.553401001 -0400
@@ -354,7 +354,7 @@ static void rtl8127_store_rss_key(struct
                 RTL_W32(tp, rss_key_reg + i, *rss_key++);
 }
 
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,8,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,8,0) || RHEL_MAJOR == 9
 int rtl8127_get_rxfh(struct net_device *dev, struct ethtool_rxfh_param *rxfh)
 {
         struct rtl8127_private *tp = netdev_priv(dev);
diff -Naurp r8127-11.014.00.orig/src/r8127_rss.h r8127-11.014.00/src/r8127_rss.h
--- r8127-11.014.00.orig/src/r8127_rss.h	2025-03-28 10:04:19.000000000 -0400
+++ r8127-11.014.00/src/r8127_rss.h	2025-04-29 12:55:30.553401001 -0400
@@ -54,7 +54,7 @@ int rtl8127_get_rxnfc(struct net_device
 int rtl8127_set_rxnfc(struct net_device *dev, struct ethtool_rxnfc *cmd);
 u32 rtl8127_get_rxfh_key_size(struct net_device *netdev);
 u32 rtl8127_rss_indir_size(struct net_device *netdev);
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,8,0)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6,8,0) || RHEL_MAJOR == 9
 int rtl8127_get_rxfh(struct net_device *dev, struct ethtool_rxfh_param *rxfh);
 int rtl8127_set_rxfh(struct net_device *dev, struct ethtool_rxfh_param *rxfh,
                      struct netlink_ext_ack *extack);
